src/main/java/com/google/finapp/SpannerDaoJDBCImpl.java

```diff
@@ -112,7 +113,8 @@ public void createCustomerRole(
     }
   }
 
-  public void moveAccountBalance(ByteArray fromAccountId, ByteArray toAccountId, BigDecimal amount)
+  public Map<ByteArray, BigDecimal> moveAccountBalance(
```

Comment:
Return ImmutableMap instead. In general: "For parameters, use the most general type that still captures the important and relevant semantic guarantees you expect (Example - List, Map). For return types, use the most specific type that you're comfortable with committing to and which explicitly reveals the semantics (Example - ImmutableList, ImmutableMap)."

Reply:
+1

@haseebajmal: I do find it a bit weird to pass a Map just for two values. I saw the long discussions on why Java doesn't provide a Pair. What would be your thoughts for using AbstractMap.SimpleImmutableEntry<BigDecimal,BigDecimal> for fromAmount, toAmount?

Reply:
Hm I think both Pair and AbstractMap.SimpleImmutableEntry<BigDecimal,BigDecimal> are semantics-free and hence make code a bit hard to read - which one is from balance and which one is to. I agree Map is weird too.

Maybe we could simply return the MoveAccountBalanceResponse directly here ? Ideally I think we would have an intermediate proto to give a good separation, but for a simple application such as this, it probably doesn't matter much.

Reply:
SG, let's go with ImmutableMap